core/dbt/graph/selector.py

```diff
@@ -258,8 +258,25 @@ def get_selected(self, spec: SelectionSpec) -> Set[UniqueId]:
                 - selectors can filter the nodes after all of them have been
                   selected
         """
-        selected_nodes = self.select_nodes(spec)
+        selected_nodes, indirect_only = self.select_nodes(spec)
         filtered_nodes = self.filter_selection(selected_nodes)
+
+        if indirect_only:
```

Comment:
It could be a neat move to wrap this logging logic in another method, defined higher up, and then call it down here, similar to `alert_non_existence`.

I also bring that up because I think the `expect_exists` property (attached to [`BaseSelectionGroup`](https://github.com/dbt-labs/dbt/blob/9af071bfe481eb41cde4e68cf9ff35b812aa629d/core/dbt/graph/selector_spec.py#L164)) could be a neat example for what we want to do around a `raise_greedy_warning`, which would raise a warning for CLI-provided selection criteria, but not for yaml selector-provided selection criteria that define `greedy: true|false` explicitly. The `alert_non_existence` method is called `if spec.expect_exists` [here](https://github.com/dbt-labs/dbt/blob/9af071bfe481eb41cde4e68cf9ff35b812aa629d/core/dbt/graph/selector.py#L149-L150).